src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+
+        try {
+            if (userFromDbOptional.isEmpty()) {
+                throw new AuthenticationException("Can't authenticate user", null);
+            }
+            User user = userFromDbOptional.get();
+            String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+            if (user.getPassword().equals(hashedPassword)) {
+                return user;
+            } else {
+                throw new AuthenticationException("Incorrect password", null);
+            }
+        } catch (Exception e) {
+            throw new AuthenticationException("Can't authenticate user", e);
+        }
```

Comment:
![image](https://github.com/mate-academy/hibernate-user-service-hw/assets/49767049/5d191c1d-b96e-42f0-a437-ab1dff6e283a)
You don't need try catch here